lib/openssl/buffering.rb

```diff
@@ -23,26 +23,20 @@ module OpenSSL::Buffering
   include Enumerable
 
   # A buffer which will retain binary encoding.
-  class Buffer < String
-    BINARY = Encoding::BINARY
-
-    def initialize
-      super
-
-      force_encoding(BINARY)
-    end
-
-    def << string
-      if string.encoding == BINARY
-        super(string)
-      else
-        super(string.b)
+  if String.method_defined?(:append_as_bytes)
+    Buffer = String
+  else
+    class Buffer
+      class Buffer < String
+        def append_as_bytes(string)
+          if string.encoding == Encoding::BINARY || string.ascii_only?
```

Comment:
Wouldn't it be better to "<< string" and force encoding again to binary at the end? I'm not familiar with ascii_only? impl, but I suspect it traverses all codepoints, and in the worst case scenario, string.b will allocate a new string  right?